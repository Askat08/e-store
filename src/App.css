/* CSS Custom Color Variables */

body {
  font-family: "Quicksand", sans-serif !important;
  background: #f3f3f3 !important;
  color: #232528 !important;
}

.navbar-brand {
  -moz-transition: all 0.25s ease-in-out;
  -webkit-transition: all 0.25s ease-in-out;
  -o-transition: all 0.25s ease-in-out;
  -webkit-transition: all 0.25s ease-in-out;
  transition: all 0.25s ease-in-out;
  opacity: 1;
  width: 65%;
}

.navbar-brand:hover {
  -webkit-transition: all 0.25s;
  -webkit-transform: scale(1.05);
  transition: all 0.25s;
  transform: scale(1.05);
  opacity: 0.7;
}

.btn-cart {
  background: transparent;
  text-transform: capitalize;
  font-size: 1rem !important;
  color: #232528;
  border-radius: 0 !important;
  border: 2px solid #5bc0de !important;
}

.btn-cart:hover {
  background: #5cb85c !important;
  color: #f3f3f3 !important;
  border: 2px solid #5cb85c !important;
}

.figure-img {
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.8),
    0 6px 20px 0 rgba(160, 137, 137, 0.19);
  cursor: pointer;
  -moz-transition: all 0.25s ease-in-out;
  -webkit-transition: all 0.25s ease-in-out;
  -o-transition: all 0.25s ease-in-out;
  -webkit-transition: all 0.25s ease-in-out;
  transition: all 0.25s ease-in-out;
  opacity: 1;
  border: 0.1rem solid rgba(185, 179, 179, 0.5);
}

.figure-img:hover {
  -webkit-transition: all 0.25s;
  -webkit-transform: scale(1.01);
  transition: all 0.25s ease-in-out;
  transform: scale(0.99);
  opacity: 0.7;
}

.not-found {
  /* visibility: hidden; */
  width: 100vw;
  padding: 1rem;
  border: solid 1rem #5bc0de;
}
