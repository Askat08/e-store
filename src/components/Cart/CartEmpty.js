import React from "react";

export default function CartEmpty() {
  return (
    <div className="conainer m-5">
      <div className="row">
        <div className="col-10 mx-auto text-center">
          <h1>Cart is empty</h1>
        </div>
      </div>
    </div>
  );
}
