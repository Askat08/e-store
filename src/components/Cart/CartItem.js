import React from "react";

export default function CartItem({ props, item }) {
  const { id, title } = item;
  const { path, extension } = item.thumbnail;
  const price = item.prices[0].price;
  const quantity = item.count;
  const https = Array.from(path);
  https.splice(4, 0, "s");
  return (
    <div className="row my-2 text-capitilize text-center">
      <div className="col-10 mx-auto col-lg-2">
        <img
          src={https.join("") + "." + extension}
          alt={title}
          className="img-fluid"
          style={{ width: "5rem", height: "5rem" }}
        />
      </div>
      <div className="col-10 mx-auto col-lg-2">
        <span className="d-lg-none">Comic : </span> {title}
      </div>
      <div className="col-10 mx-auto col-lg-2">
        <span className="d-lg-none">Price : </span>
        <strong>{`$${price.toFixed(2)}`}</strong>
      </div>
      <div className="col-10 mx-auto col-lg-2 my-2 my-lg-0">
        <div className="d-flex justify-content-center">
          <div className="">
            <span
              className="btn btn-cart mx-1"
              onClick={() => props.decrementCount(id)}
            >
              -
            </span>
            <span className="btn btn-cart mx-1">{quantity}</span>
            <span
              className="btn btn-cart mx-1"
              onClick={() => props.incrementCount(id)}
            >
              +
            </span>
          </div>
        </div>
      </div>

      <div className="col-10 mx-auto col-lg-2">
        <div
          style={{ cursor: "pointer" }}
          onClick={() => {
            props.removeItemFromCart(id, props.cart);
          }}
        >
          <i className="fas fa-trash text-danger" />
        </div>
      </div>

      <div className="col-10 mx-auto col-lg-2">
        <strong>Item total : $ {(price * quantity).toFixed(2)}</strong>
      </div>
    </div>
  );
}
