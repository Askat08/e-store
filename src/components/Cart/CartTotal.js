import React from "react";
import { Link } from "react-router-dom";

export default function CartTotal({ props }) {
  const { cartTax, cartTotal, cartSubTotal, handleClearCart } = props;

  return (
    <React.Fragment>
      <div className="container">
        <div className="row">
          <div className="col-10 mt-2 ml-sm-5 ml-md-auto col-sm-8 text-right text-capitilize">
            <Link to="/">
              <button
                className="btn btn-outline-danger text-uppercase mb-3 px-5"
                type="button"
                onClick={() => handleClearCart()}
              >
                clear cart
              </button>
            </Link>
            <h5>
              <span className="text-info">subtotal : </span>
              <strong>$ {cartSubTotal}</strong>
            </h5>
            <h5>
              <span className="text-info">tax* : </span>
              <strong>$ {cartTax}</strong>
            </h5>
            <h5>
              <span className="text-info">total : </span>
              <strong>$ {cartTotal}</strong>
            </h5>
            <br />
            <br />
            <br />
            <br />
            <p>*New York State Sales Tax 8.875%</p>
          </div>
        </div>
      </div>
    </React.Fragment>
  );
}
