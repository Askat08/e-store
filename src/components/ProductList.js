import React from "react";
import Product from "./Product";
import SearchBox from "./SearchBox";

const ProductList = (props) => {
  const {
    items,
    handleAddtoCart,
    messageAlert,
    isError,
    onSearchChange,
    onSearchClick,
    searchfield,
  } = props;

  const comics = items
    .map((comic, i) => {
      return (
        <Product
          key={comic.id}
          id={comic.id}
          price={comic.prices[0].price}
          title={comic.title}
          path={comic.thumbnail.path}
          ext={comic.thumbnail.extension}
          description={comic.description}
          handleAddtoCart={handleAddtoCart}
          messageAlert={messageAlert}
        />
      );
    })
    .slice(0, 15); // render only first 15 items

  // checking if items exists to render and no error

  if (items.length === 0 && !isError) {
    return (
      <div className="d-flex align-items-center justify-content-center">
        <div className="">
          <SearchBox
            onSearchChange={onSearchChange}
            searchField={searchfield}
            onSearchClick={onSearchClick}
          />
        </div>
        <strong>Loading...</strong>
        <div
          className="spinner-border m-5"
          role="status"
          aria-hidden="true"
        ></div>
      </div>
    );
  } else if (isError) {
    return (
      <div>
        <p className="lead text-center">Data not found</p>
        <SearchBox onSearchChange={onSearchChange} searchField={searchfield} />
      </div>
    );
  } else {
    return (
      <React.Fragment>
        <div className="jumbotron m-0">
          <div className="">
            <SearchBox
              onSearchChange={onSearchChange}
              searchField={searchfield}
              onSearchClick={onSearchClick}
            />
          </div>
          <div className="row m-5 justify-content-center">{comics}</div>
        </div>
      </React.Fragment>
    );
  }
};

export default ProductList;
