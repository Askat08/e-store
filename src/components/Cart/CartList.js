import React from "react";
import CartItem from "./CartItem";

export default function CartList({ cart, props }) {
  const filteredCart = cart.filter((v, i, a) => a.indexOf(v) === i);
  return (
    <div className="container-fluid">
      {filteredCart.map((item) => {
        return <CartItem key={item.id} item={item} props={props} />;
      })}
    </div>
  );
}
