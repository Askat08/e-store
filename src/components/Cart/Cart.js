import React from "react";
import CartData from "./CartData";
import CartEmpty from "./CartEmpty";
import CartList from "./CartList";
import CartTotal from "./CartTotal";

const Cart = (props) => {
  if (props.cart.length > 0) {
    const { cart } = props;
    return (
      <div>
        <h3 className="m-5 text-center">Comics in cart</h3>
        <CartData />
        <CartList cart={cart} props={props} />
        <CartTotal props={props} />
      </div>
    );
  } else {
    return <CartEmpty />;
  }
};

export default Cart;
