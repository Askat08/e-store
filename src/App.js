import React, { Component } from "react";
import { Switch, Route } from "react-router-dom";
import { BrowserRouter as Router } from "react-router-dom";
import "./App.css";

// Using Bootstrap
import "bootstrap/dist/css/bootstrap.min.css";

// Importing Components
import Navbar from "./components/Navbar";
import ProductList from "./components/ProductList";
import Cart from "./components/Cart/Cart";
import NotFound from "./components/NotFound";
import Footer from "./components/Footer";
import Alert from "./components/Alert";

import { cartItemModifyCount, setLocalStorage } from "./utility/functions";

class App extends Component {
  constructor(props) {
    super(props);
    const filteredCart = JSON.parse(localStorage.getItem("cart-items"));

    this.state = {
      items: [],
      isError: false,
      searchfield: "",
      cart: filteredCart || [],
      cartSubTotal: 0,
      cartTotal: 0,
      cartTax: 0,
      flashAlert: false,
      attributionHTML: "",
    };
  }

  // Alert message when item added to cart
  flashAlert = () => {
    this.setState(
      () => {
        return {
          flashAlert: !this.state.flashAlert,
        };
      },
      () => {
        setTimeout(() => {
          this.setState({
            flashAlert: !this.state.flashAlert,
          });
        }, 500);
      }
    );
  };

  componentDidMount() {
    // Set initial state
    let url = `https://gateway.marvel.com/v1/public/comics?orderBy=modified&ts=1&apikey=0860ac3aed33051450d554be9f0d84d2&hash=c209f33ace43013413284d09f7e06b6e`;

    this.fetchData(url);

    this.countTotalPrice(this.state.cart);
  }

  // Fetch data from api
  fetchData = (url) => {
    return fetch(url)
      .then((res) => res.json())
      .then((data) =>
        this.setState(() => {
          const newComics = data.data.results;
          return {
            items: newComics.concat(this.state.items),
            attributionHTML: data.attributionHTML,
            // cart: data.data.results,
          };
        })
      )
      .catch((err) => this.setState({ isError: !this.state.isError }));
  };

  // handles when search input value changes
  onSearchChange = (e) => {
    this.setState({ searchfield: e.target.value });
  };

  // handles when the search input value is submited
  onSearchClick = (e) => {
    if (e.key === "Enter" || e.detail === 1) {
      if (this.state.searchfield) {
        let url = `https://gateway.marvel.com/v1/public/comics?titleStartsWith=${this.state.searchfield}&limit=15&ts=1&apikey=0860ac3aed33051450d554be9f0d84d2&hash=c209f33ace43013413284d09f7e06b6e`;
        this.fetchData(url);
      }
    }
  };

  // Handles total price of items in cart
  countTotalPrice = (arr) => {
    const cart = [...arr];
    // Gets total price of cart's items
    const total = cart
      .filter((item, index) => cart.indexOf(item) === index)
      .reduce((acc, item) => item.count * item.prices[0].price + acc, 0)
      .toFixed(2);

    const tax = (0.08875 * total).toFixed(2);
    const totalWithTax = (parseFloat(total) + parseFloat(tax)).toFixed(2);

    this.setState(() => {
      return {
        cartSubTotal: total,
        cartTax: tax,
        cartTotal: totalWithTax,
      };
    });
  };

  // handles decrementing quantity count of item added to cart
  decrementCount = (id) => {
    const newValue = cartItemModifyCount(
      this.state.cart,
      id,
      "dec",
      this.state.cart
    );

    this.setState(() => {
      return {
        cart: newValue,
      };
    });
    this.countTotalPrice(this.state.cart);
  };

  // handles incrementing quantity count of item added to cart
  incrementCount = (id) => {
    const newValue = cartItemModifyCount(
      this.state.cart,
      id,
      "inc",
      this.state.cart
    );

    this.setState(() => {
      return {
        cart: newValue,
      };
    });
    this.countTotalPrice(this.state.cart);
  };

  // handles removing Item from cart
  removeItemFromCart = (id, arr) => {
    const cart = [...arr];
    const removedItem = cart.filter((item) => item.id === id);
    // setting count of removed item to 0
    removedItem[0].count = 0;

    this.setState(
      () => {
        return {
          cart: cart.filter((item) => item.id !== id),
        };
      },
      () => {
        this.countTotalPrice(this.state.cart);
      }
    );
  };

  // Method handle adding item to cart
  handleAddtoCart = (id, cartArr, itemsArr) => {
    const added = cartItemModifyCount(itemsArr, id, "add", cartArr);

    this.setState(
      () => {
        return {
          cart: added, //duplicates added after  refresh the page ???
        };
      },
      () => {
        this.countTotalPrice(this.state.cart);
        setLocalStorage(this.state.cart);
      }
    );
    this.flashAlert();
  };

  // Handle resetting cart items
  handleClearCart = (e) => {
    this.setState(
      () => {
        return {
          cart: [],
        };
      },
      () => {
        localStorage.clear();
      }
    );
  };

  render() {
    return (
      <Router basename={process.env.PUBLIC_URL}>
        <Navbar />
        {this.state.flashAlert ? <Alert /> : null}
        <Switch>
          <Route
            path="/"
            exact
            render={(props) => (
              <ProductList
                {...this.state}
                onSearchChange={this.onSearchChange}
                onSearchClick={this.onSearchClick}
                handleAddtoCart={this.handleAddtoCart}
              />
            )}
          />
          <Route
            path="/cart"
            render={(props) => (
              <Cart
                {...this.state}
                decrementCount={this.decrementCount}
                incrementCount={this.incrementCount}
                removeItemFromCart={this.removeItemFromCart}
                handleClearCart={this.handleClearCart}
              />
            )}
          />
          <Route component={NotFound} />
        </Switch>
        <Footer props={this.state.attributionHTML} />
      </Router>
    );
  }
}

export default App;
