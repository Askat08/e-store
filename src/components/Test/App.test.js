import React from "react";
import { shallow, mount } from "enzyme";

import App from "../../App";
import Cart from "../Cart/Cart";
import ProductList from "../ProductList";
import * as methods from "../../utility/functions";

describe("App container test", () => {
  const cartMock = [
    { title: "thor", id: 1, count: 1, prices: [{ price: 2.99 }] },
    { title: "loki", id: 2, count: 1, prices: [{ price: 2.25 }] },
    { title: "hulk", id: 3, count: 1, prices: [{ price: 23.99 }] },
    { title: "spider-man", id: 4, count: 1, prices: [{ price: 15.19 }] },
  ];

  const itemsMock = [
    { title: "iron man", id: 5, count: 1, prices: [{ price: 2.99 }] },
    { title: "deadpool", id: 6, count: 1, prices: [{ price: 2.25 }] },
    { title: "daredevil", id: 7, count: 1, prices: [{ price: 23.99 }] },
    { title: "captain America", id: 8, count: 1, prices: [{ price: 15.19 }] },
  ];

  let wrapper, instance;

  beforeEach(() => {
    wrapper = shallow(<App />);
    instance = wrapper.instance(); // assign instance of the wrapper
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("should render app container", () => {
    const wrapper = mount(<App />);
    expect(wrapper).toMatchSnapshot();
  });

  // it("should call flashAlert when handleAddtoCart called", () => {
  //   instance.flashAlert = jest.fn();

  //   jest
  //     .spyOn(instance, "handleAddtoCart")
  //     .mockImplementation(() => instance.flashAlert());

  //   wrapper.update();
  //   instance.handleAddtoCart();
  //   expect(instance.flashAlert).toBeCalled();
  //   instance.handleAddtoCart.mockRestore();
  // });

  it("state changes when flashAlert method is called", () => {
    wrapper.instance().flashAlert();
    expect(instance.state.flashAlert).toBeTruthy();
    wrapper.instance().flashAlert();
    expect(instance.state.flashAlert).toBeFalsy();
  });

  it("state changes when onSearchChange method is called", () => {
    jest
      .spyOn(instance, "onSearchChange")
      .mockImplementation((text) => wrapper.setState({ searchfield: text }));

    wrapper.update();
    wrapper.instance().onSearchChange("thor");
    expect(instance.state.searchfield).toBe("thor");
    instance.onSearchChange.mockRestore();
  });

  it("state changes when onSearchChange method is called", () => {
    const e = {
      target: {
        value: "thor",
      },
    };
    jest.spyOn(instance, "onSearchChange");
    wrapper.instance().onSearchChange(e);
    expect(instance.state.searchfield).toEqual("thor");
    expect(instance.onSearchChange).toHaveBeenCalled();
  });

  // it("should run methods with proper arguments when `componentDidMount()`", () => {
  //   jest.spyOn(instance, "countTotalPrice");
  //   jest.spyOn(instance, "fetchData");
  //   instance.componentDidMount();
  //   expect(instance.countTotalPrice).toHaveBeenCalledTimes(1);
  //   expect(instance.fetchData).toHaveBeenCalledTimes(1);
  //   expect(instance.fetchData).toHaveBeenCalledWith(expect.any(String));
  //   instance.countTotalPrice.mockRestore();
  //   instance.fetchData.mockRestore();
  // });

  it("fetch data when`componentDidMount()`", () => {
    jest
      .spyOn(instance, "fetchData")
      .mockImplementation(() =>
        wrapper.setState({ items: ["thor", "loki", "hulk"] })
      );
    wrapper.update();
    instance.componentDidMount();
    expect(instance.fetchData).toBeCalled();
    expect(instance.state.items).toContain("hulk");
    expect(instance.state.items).toContain("loki");
    expect(instance.state.items).toContain("thor");
    expect(instance.state.items).toHaveLength(3);
    instance.fetchData.mockRestore();
  });

  it("fetch data by name when searchClick()", () => {
    jest.spyOn(instance, "onSearchClick").mockImplementation((key, text) => {
      wrapper.setState({ searchfield: text });
      if (key === "Enter" || key === 1) {
        if (instance.state.searchfield) {
          return instance.fetchData(instance.state.searchfield);
        }
      }
    });
    jest.spyOn(instance, "fetchData");
    instance.onSearchClick("Enter", "thor");
    expect(instance.fetchData).toBeCalledWith("thor");
    expect(instance.fetchData).toHaveBeenCalledTimes(1);
    instance.onSearchClick.mockRestore();
    instance.fetchData.mockRestore();
  });

  it("checking if key or detail is correct when searchClick()", () => {
    jest.spyOn(instance, "fetchData");
    const e = {
      key: "Enter",
      detail: 1,
    };
    instance.setState({ searchfield: "thor" });
    instance.onSearchClick(e);
    expect(instance.fetchData).toBeCalled();
  });

  it("gets item by id", () => {
    expect(methods.getItembyId(1, cartMock)).toEqual({
      title: "thor",
      id: 1,
      count: 1,
      prices: [{ price: 2.99 }],
    });
    expect(methods.getItembyId(4, cartMock)).toEqual({
      title: "spider-man",
      id: 4,
      count: 1,
      prices: [{ price: 15.19 }],
    });
  });

  it("should count total price of items in arr", () => {
    instance.countTotalPrice(cartMock);
    expect(instance.state.cartSubTotal).toEqual("44.42");
    expect(instance.state.cartTax).toEqual("3.94");
    expect(instance.state.cartTotal).toEqual("48.36");
  });

  it("should modify count of item in an array", () => {
    let newArr = [];
    newArr = methods.cartItemModifyCount(cartMock, 2, "inc", cartMock);
    expect(newArr[1].count).toEqual(2);
    newArr = methods.cartItemModifyCount(cartMock, 3, "inc", cartMock);
    expect(newArr[2].count).toEqual(2);
    newArr = methods.cartItemModifyCount(cartMock, 2, "inc", cartMock);
    expect(newArr[1].count).toEqual(3);
    newArr = methods.cartItemModifyCount(cartMock, 2, "inc", cartMock);
    expect(newArr[1].count).toEqual(4);
    newArr = methods.cartItemModifyCount(cartMock, 2, "dec", cartMock);
    expect(newArr[1].count).toEqual(3);
    newArr = methods.cartItemModifyCount(cartMock, 2, "add", cartMock);
    expect(newArr[1].count).toEqual(4);
  });

  it("should increment and count total price of items in array", () => {
    const countModifySpy = jest.spyOn(methods, "cartItemModifyCount");
    countModifySpy.mockReturnValue(
      methods.cartItemModifyCount(cartMock, 3, "inc", cartMock)
    );
    const countPriceSpy = jest.spyOn(instance, "countTotalPrice");
    countPriceSpy.mockReturnValue(instance.countTotalPrice(cartMock));
    instance.incrementCount();
    expect(instance.state.cart[2].count).toEqual(3);
    expect(instance.state.cartSubTotal).toEqual("99.15");
    expect(instance.countTotalPrice).toHaveBeenCalled();
    countModifySpy.mockRestore();
    countPriceSpy.mockRestore();
  });

  it("should decrement and count total price of items in array", () => {
    const decountModifySpy = jest.spyOn(methods, "cartItemModifyCount");
    decountModifySpy.mockReturnValue(
      methods.cartItemModifyCount(cartMock, 2, "dec", cartMock)
    );
    const countPriceSpy = jest.spyOn(instance, "countTotalPrice");
    countPriceSpy.mockReturnValue(instance.countTotalPrice(cartMock));
    instance.decrementCount();
    expect(instance.state.cart[1].count).toEqual(3);
    expect(instance.state.cartSubTotal).toEqual("96.90");
    decountModifySpy.mockRestore();
    countPriceSpy.mockRestore();
  });

  it("remove item by ID from array", () => {
    instance.removeItemFromCart(2, cartMock);
    expect(instance.state.cart).toHaveLength(3);
  });

  it("add item to cart and flash alert message", () => {
    jest.spyOn(instance, "flashAlert");
    jest.spyOn(instance, "countTotalPrice");
    jest.spyOn(methods, "setLocalStorage");

    const handleAddtoCartSpy = jest.spyOn(instance, "handleAddtoCart");
    handleAddtoCartSpy.mockImplementation(() => {
      const added = methods.cartItemModifyCount(itemsMock, 6, "add", cartMock);
      instance.countTotalPrice(added);
      methods.setLocalStorage(added);
      instance.flashAlert();

      instance.setState({
        cart: added,
      });
    });

    const addedItem = [
      {
        title: "deadpool",
        id: 6,
        count: 2,
        prices: [{ price: 2.25 }],
      },
    ];

    wrapper.update();
    instance.handleAddtoCart();

    expect(instance.state.cart).toHaveLength(5);
    expect(instance.state.cart).toEqual([...cartMock, ...addedItem]);
    expect(instance.state.flashAlert).toBeTruthy();

    expect(instance.countTotalPrice).toHaveBeenCalledWith([
      ...cartMock,
      ...addedItem,
    ]);

    expect(instance.flashAlert).toHaveBeenCalled();
    expect(methods.setLocalStorage).toHaveBeenCalled();

    handleAddtoCartSpy.mockReset();
  });

  it("will empty cart on handleClearCart()", () => {
    instance.handleClearCart();
    expect(instance.state.cart).toEqual([]);
    instance.setState({ cart: [1, 2, 3, 4, 5, 6, 7, 78] });
    expect(instance.state.cart).toHaveLength(8);
    instance.handleClearCart();
    expect(instance.state.cart).toEqual([]);
  });

  it("Route has proper props", () => {
    expect(wrapper.find("Route").at(2).props().component).toEqual(
      expect.any(Function)
    );
    expect(wrapper.find("Route").at(0).props()).toEqual({
      path: "/",
      exact: true,
      render: expect.any(Function),
    });
    expect(wrapper.find("Route").at(1).props()).toEqual({
      path: "/cart",
      render: expect.any(Function),
    });
  });

  it("ProductList has proper props", () => {
    const mockProps = {
      attributionHTML: "",
      cart: [],
      cartSubTotal: "0.00",
      cartTax: "0.00",
      cartTotal: "0.00",
      flashAlert: false,
      handleAddtoCart: expect.any(Function),
      isError: false,
      items: [],
      onSearchChange: expect.any(Function),
      onSearchClick: expect.any(Function),
      searchfield: "",
    };

    expect(wrapper.find("Route").at(0).props().render()).toEqual(
      <ProductList {...mockProps} />
    );
  });

  it("Cart has proper props", () => {
    const mockProps = {
      attributionHTML: "",
      cart: [],
      cartSubTotal: "0.00",
      cartTax: "0.00",
      cartTotal: "0.00",
      flashAlert: false,
      isError: false,
      items: [],
      searchfield: "",
      decrementCount: expect.any(Function),
      incrementCount: expect.any(Function),
      removeItemFromCart: expect.any(Function),
      handleClearCart: expect.any(Function),
    };

    expect(wrapper.find("Route").at(1).props().render()).toEqual(
      <Cart {...mockProps} />
    );
  });
});
