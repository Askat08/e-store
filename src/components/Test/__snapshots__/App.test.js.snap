// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App container test should render app container 1`] = `
<App>
  <BrowserRouter
    basename=""
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Navbar>
        <nav
          className="navbar navbar-expand-sm bg-dark   navbar-dark"
        >
          <Link
            to="/"
          >
            <LinkAnchor
              href="/"
              navigate={[Function]}
            >
              <a
                href="/"
                onClick={[Function]}
              >
                <img
                  alt="store-bag-logo"
                  className="navbar-brand"
                  src="logo.svg"
                />
              </a>
            </LinkAnchor>
          </Link>
          <ul
            className="navbar-nav align-items-center"
          >
            <li
              className="nav-item ml-5"
            >
              <Link
                className="nav-link text-info"
                to="/"
              >
                <LinkAnchor
                  className="nav-link text-info"
                  href="/"
                  navigate={[Function]}
                >
                  <a
                    className="nav-link text-info"
                    href="/"
                    onClick={[Function]}
                  >
                    <h4>
                      comics
                    </h4>
                  </a>
                </LinkAnchor>
              </Link>
            </li>
          </ul>
          <Link
            className="ml-auto"
            to="/cart"
          >
            <LinkAnchor
              className="ml-auto"
              href="/cart"
              navigate={[Function]}
            >
              <a
                className="ml-auto"
                href="/cart"
                onClick={[Function]}
              >
                <button
                  className="btn btn-info"
                >
                  <i
                    className="fas fa-cart-arrow-down"
                  />
                               
                  My Cart
                </button>
              </a>
            </LinkAnchor>
          </Link>
        </nav>
      </Navbar>
      <Switch>
        <Route
          computedMatch={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/",
              "url": "/",
            }
          }
          exact={true}
          location={
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            }
          }
          path="/"
          render={[Function]}
        >
          <ProductList
            attributionHTML=""
            cart={Array []}
            cartSubTotal="0.00"
            cartTax="0.00"
            cartTotal="0.00"
            flashAlert={false}
            handleAddtoCart={[Function]}
            isError={false}
            items={Array []}
            onSearchChange={[Function]}
            onSearchClick={[Function]}
            searchfield=""
          >
            <div
              className="d-flex align-items-center justify-content-center"
            >
              <div
                className=""
              >
                <SearchBox
                  onSearchChange={[Function]}
                  onSearchClick={[Function]}
                  searchField=""
                >
                  <div
                    className="input-group col-sm-9 mx-auto"
                  >
                    <input
                      className="form-control py-2 shadow-none"
                      id="example-search-input"
                      onChange={[Function]}
                      onKeyPress={[Function]}
                      placeholder="Search comic book..."
                      type="search"
                      value=""
                    />
                    <span
                      className="input-group-append"
                    >
                      <button
                        className="btn btn-outline-info shadow-none"
                        onClick={[Function]}
                        type="button"
                      >
                        <i
                          className="fa fa-search"
                        />
                      </button>
                    </span>
                  </div>
                </SearchBox>
              </div>
              <strong>
                Loading...
              </strong>
              <div
                aria-hidden="true"
                className="spinner-border m-5"
                role="status"
              />
            </div>
          </ProductList>
        </Route>
      </Switch>
      <Footer
        props=""
      >
        <div
          className="sticky-bottom"
        >
          <div
            className="card-footer bg-dark text-light"
          >
            <div
              className="row"
            >
              <h3
                className="col-6 my-3"
              >
                Marvel Comic Books
              </h3>
              <h4
                className="card-text my-3 col-6 text-right"
              >
                Get your superhero today.
              </h4>
            </div>
            <div
              className="row"
            >
              <h5
                className="col-6 my-3 "
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "",
                  }
                }
              />
              <h5
                className="col-6 my-3 text-right"
              >
                Coded with 
                <i
                  className="fas fa-laptop-code text-info"
                />
                 by
                 
                <a
                  className="text-danger"
                  href="https://askatb.com"
                  target="_blank"
                >
                  me
                </a>
                .
              </h5>
            </div>
            <button
              className="btn btn-outline-info shadow-none"
              onClick={[Function]}
              type="button"
            >
              top
            </button>
          </div>
        </div>
      </Footer>
    </Router>
  </BrowserRouter>
</App>
`;
